tests/unit/imgix-component.spec.ts

```diff
@@ -1,54 +1,74 @@
+import VueImgix from '@/plugins/vue-imgix/index';
+import { IxImg } from '@/plugins/vue-imgix/ix-img';
+import { IxPicture } from '@/plugins/vue-imgix/ix-picture';
+import { IxSource } from '@/plugins/vue-imgix/ix-source';
+import { render } from '@testing-library/vue';
+import { config, mount } from '@vue/test-utils';
 import { createApp } from 'vue';
-import VueImgix, { IxImg } from '@/plugins/vue-imgix';
-import '@testing-library/jest-dom';
-import { render, screen } from '@testing-library/vue';
 import _App from '../../src/App.vue';
 import {
   expectElementToHaveFixedSrcAndSrcSet,
   expectElementToHaveFluidSrcAndSrcSet,
 } from '../helpers/url-assert';
-
-const App = createApp(_App);
+/**
+ * Why register the plugin and each individual component globally?
+ *
+ * It's a limitation with  @vue/test-utils at the moment. The
+ * `config.global.plugins` option is supposed to register plugins globally,
+ * but it doesn't seem to work when tests are run in parallel.
+ *
+ * The only reliable way to register plugins components globally is to register
+ * them individually as `config.global.components` as well as
+ * `config.global.plugins`.
+ */
+config.global.plugins = [[VueImgix, { domain: 'assets.imgix.net' }]];
+config.global.components = {
+  IxImg,
+  IxPicture,
+  IxSource,
+};
 
 describe('imgix component', () => {
-  beforeAll(() => {
-    App.use(VueImgix, {
-      domain: 'assets.imgix.net',
-    });
-  });
   it('an img should be rendered', () => {
-    render(IxImg, {
+    const wrapper = mount(IxImg, {
       props: {
         src: 'examples/pione.jpg',
-        // TODO(luis): remove this when we have a better way of testing
         ['data-testid']: 'img-rendering',
       },
     });
 
-    expect(screen.getByTestId('img-rendering'));
+    const el = wrapper.find('img[data-testid="img-rendering"]').html();
+
+    expect(el).toBeTruthy();
   });
   it(`the rendered img's src should be set`, () => {
-    render(IxImg, {
+    const wrapper = mount(IxImg, {
       props: {
         src: 'examples/pione.jpg',
         ['data-testid']: 'img-rendering',
       },
     });
 
-    expect(screen.getByTestId('img-rendering')).toHaveAttribute(
-      'src',
-      expect.stringMatching('examples/pione.jpg'),
-    );
+    const el = wrapper.get('img[data-testid="img-rendering"]').html();
+    // create HTMLElement from el
+    const img = new DOMParser().parseFromString(el, 'text/html').body
+      .firstChild as HTMLImageElement;
+    const srcAttr = img.getAttribute('src');
```

Comment:
This seems awfully roundabout - is there no better way to just get the `src` from an element? e.g. from below, using `element.getAttribute`?

Reply:
This was essentially my :flip_table: in code form. I could not for the life for me do `wrapper.get('some-ele').getAttribute('some-attr') to work for this test. 

```console
Property 'getAttribute' does not exist on type 'Omit<DOMWrapper<Element>, "exists">'. Did you mean 'attributes'?
```

Turns out like you suggested it's just a stupid mistake on my part. The method for `@vue/test-utils` wrappers is slightly different.

```ts
    const img = wrapper.get('img[data-testid="img-rendering"]').element;
    const srcAttr = img.getAttribute('src');
```